### PR TITLE
[#338] As a developer, I can sync signing by match with Fastlane Swift

### LINKED ISSUE #338: As a developer, I can sync signing by match with Fastlane Swift
## Why

Currently, iOS developers need to learn at least basic Ruby to understand or write/update Fastlane configurations. To reduce the effort, we will create Fastlane in Swift language. So everyone in the iOS Chapter can use easier.

## Acceptance Criteria

- Use Fastlane's [match](https://docs.fastlane.tools/actions/match/#match) action to sync signing for development, ad-hoc, appstore.

### PR DESCRIPTION
**Pull request by suho**
Resolves #338

## What happened

- I added Match and Keychain helpers to support sync code signing by math
 
## Insight

- Currently, we can use Xcode to write custom Fastlane code 🙌
- With this pull request, now we can support `match` with swift.
- How to use this:
```swift
// Add values for constant and secret, then
Match.syncCodeSigning(type: .development, appIdentifier: ["co.nimblehq.ios.templates"], isForce: true)
Match.syncCodeSigning(type: .adHoc, appIdentifier: ["co.nimblehq.ios.templates"])
Match.syncCodeSigning(type: .appStore, appIdentifier: ["co.nimblehq.ios.templates"])
```

Note: I did not define bundleId yet, will do it later.
 
## Proof Of Work

AppStore:
Screen Shot 2022-09-22 at 17 13 26

For Development and Adhoc profiles, you can check it here: https://github.com/nimblehq/match-certificates/tree/master/profiles

### CODE REVIEW

File: fastlane/Helpers/Match.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }
+    }
+}
+
+extension Match {
+
+    enum MatchType: String {
+
+        case development
+        case adHoc = "adhoc"
```
Comment (ducbm051291) [inline fastlane/Helpers/Match.swift:44]: I think we dont have to use camel case for `adHoc` and `appStore`. Or is there any reason behind this?

**Review by ducbm051291 — APPROVED** [review]
The rest LGTM 💯

**Comment by blyscuit** [comment]
You can comment out the existing fastlane script for this epic 🙏

File: fastlane/Fastfile.swift [hunk 2]
```diff
@@ -9,8 +9,11 @@
...
 class Fastfile: LaneFile {
-	func customLane() {
-	desc("Description of what the lane does")
-		// add actions here: https://docs.fastlane.tools/actions
+
+	func syncCodeSigningLane() {
+        desc("This lane is for development purpose, will be removed after the migration")
+        Match.syncCodeSigning(type: .development, appIdentifier: ["co.nimblehq.ios.templates"])
+        Match.syncCodeSigning(type: .adHoc, appIdentifier: ["co.nimblehq.ios.templates"])
+        Match.syncCodeSigning(type: .appStore, appIdentifier: ["co.nimblehq.ios.templates"])
```
Comment (blyscuit) [inline fastlane/Fastfile.swift:17]: this indent looks weird to me 🤔

File: fastlane/Helpers/Match.swift (hunk 1 above)
Comment (suho) [reply fastlane/Helpers/Match.swift:44]: @ducbm051291 We should use English when defining enum cases.
So we have:
- Ad hoc
- App Store

File: fastlane/Constants/Constant.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//
+//  Constant.swift
+//  FastlaneRunner
+//
+//  Created by Su Ho on 22/09/2022.
+//  Copyright © 2022 Joshua Liebowitz. All rights reserved.
```
Comment (blyscuit) [inline fastlane/Constants/Constant.swift:6]: I think we should modify the Copyright to Nimble when we are editing these files for the first time.

**Review by blyscuit — COMMENTED** [review]
Looks great

File: fastlane/Helpers/Match.swift (hunk 1 above)
Comment (blyscuit) [reply fastlane/Helpers/Match.swift:44]: `match` is the only one that use `ad hoc` as `adhoc`  😆

File: fastlane/Fastfile.swift (hunk 2 above)
Comment (suho) [reply fastlane/Fastfile.swift:17]: What do you mean? I think I will change from tabs to spaces :D

File: fastlane/Constants/Constant.swift (hunk 3 above)
Comment (suho) [reply fastlane/Constants/Constant.swift:6]: Good catch! Let's me change the organization on the Xcode

**Review by blyscuit — APPROVED** [review]

**Review by nkhanh44 — APPROVED** [review]
🚀

**Review by phongvhd93 — APPROVED** [review]

**Comment by markgravity** [comment]
What is the plan for this `migrate-develop`? 🤔

**Comment by suho** [comment]
@markgravity We will create a PR to merge `migrate-develop` to `develop` branch after the Nimble Growth #40. So I think it's good to add the milestone to `3.11.0` instead.
Could you help me to create a new milestone for `3.11.0`

**Review by vnntsu — APPROVED** [review]
LGTM! 🎉

**Pull request merged**